test/test_profiler.py

```diff
@@ -1184,28 +1193,71 @@ def end_time_ns(self):
             def name(self) -> str:
                 return self._name
 
-        cuda_events = [
-            MockKinetoEvent("cudaLaunchKernel", 400, 100, 1, DeviceType.CPU),
-            MockKinetoEvent("cudaLaunchKernel", 500, 100, 2, DeviceType.CPU),
-            MockKinetoEvent("cudaLaunchKernel", 600, 100, 3, DeviceType.CPU),
-            MockKinetoEvent("GPU", 700, 100, 1, DeviceType.CUDA),
-            MockKinetoEvent("GPU", 800, 100, 2, DeviceType.CUDA),
-            MockKinetoEvent("GPU", 900, 100, 3, DeviceType.CUDA)
-        ]
+        if cuda_events_json == "":
+            cuda_events = [
+                MockKinetoEvent("cudaLaunchKernel", 400, 100, 1, 0),
+                MockKinetoEvent("cudaLaunchKernel", 500, 100, 2, 0),
+                MockKinetoEvent("cudaLaunchKernel", 600, 100, 3, 0),
+                MockKinetoEvent("cudaLaunchKernel", 700, 100, 4, 0),
+                MockKinetoEvent("cudaLaunchKernel", 800, 100, 5, 0),
+                MockKinetoEvent("cudaLaunchKernel", 1500, 100, 6, 0),
+                MockKinetoEvent("GPU", 900, 100, 1, 1),
+                MockKinetoEvent("GPU", 1000, 100, 2, 1),
+                MockKinetoEvent("GPU", 1100, 100, 3, 1),
+                MockKinetoEvent("GPU", 1200, 100, 4, 1),
+                MockKinetoEvent("GPU", 1300, 100, 5, 1),
+                MockKinetoEvent("GPU", 1700, 100, 6, 1)
+            ]
+        else:
+            json_list = json.loads(cuda_events_json)
```

Comment:
I think it would make sense to pull the loading code into a separate class. (That way you could also bundle it with the serialization code.) Right now the flow is a little awkward, because the structure is:
```
[Inline definition]
[Load from file]
[Inline definition]
[Load from file]
```